internal/server/http/response_handler.go

```diff
@@ -54,14 +59,20 @@ func (rh *ResponseHandler) fillDefaults(imposter *Imposter) {
 		rh.counter = 1
 		rh.currentInd = 0
 	}
+
+	rh.mutex = new(sync.Mutex)
```

Comment:
AFAIK, there's no need to make the `rh.mutex` a pointer neither to initialize it since `rh` is already a pointer to a `ResponseHandler`. See [this example](https://tour.golang.org/concurrency/9) from the Go Tour. Thanks!